src/prowjobsscraper/equinix_usages.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from datetime import datetime
+from typing import Final, Optional
+
+import requests
+from pydantic import BaseModel
+
+logger = logging.getLogger(__name__)
+
+
+class EquinixUsage(BaseModel):
+    description: Optional[str]
+    facility: str
+    metro: str
+    name: str
+    plan: str
+    plan_version: str
+    price: float
+    quantity: float
+    total: float
+    type: str
+    instance: Optional[str]
+    unit: str
+    start_date: datetime
+    end_date: Optional[datetime]
+
+    @property
+    def job_build_id(self) -> str:
+        return self.name.split("-")[-1]
+
+
+class EquinixUsageEvent(BaseModel):
+    build_id: str
+    usage: EquinixUsage
+
+    @classmethod
+    def create_from_equinix_usage(cls, usage: EquinixUsage) -> "EquinixUsageEvent":
+        return cls(build_id=usage.job_build_id, usage=usage)
+
+
+class EquinixUsagesExtractor:
+    """
+    EquinixUsagesExtractor parses the Equinix usages data gathered from equinix metal API.
+    """
+
+    _EQUINIX_ENDPOINT_HEADER: Final[str] = "X-Auth-Token"
+    _EQUINIX_METAL_ENDPOINT: Final[
+        str
+    ] = "https://api.equinix.com/metal/v1/projects/{}/usages"
+
+    def __init__(self, project_id: str, project_token: str):
+        self._project_id = project_id
+        self._project_token = project_token
+
+    def get_project_usages(
+        self,
+    ) -> list[EquinixUsage]:
+        equinix_project_usages = requests.get(
```

Comment:
No, a lot of things from the docs behave different than expected

Reply:
what is the time period covered by the call to the API? Could be good to add a comment with this information and/or a link to the API documentation.

Reply:
It is currently retrieving the past month events, i.e. if you make the call on 27.3, you will get all March usages up until the call

Reply:
so that means on April 1st at 00:00, we loose the data of the time period March 31st, 23:00 to April 1st, 00:00?

Reply:
I believe so, it is different than what is written in the docs, but that is the behavior

Reply:
every 1h, but we should not take this information into consideration as it is configurable

Reply:
there is no way in equinix api to get data from the previous month?

Reply:
OK I can get it to work now hopefully

Reply:
Do we prefer it to take the last month's usages?

Reply:
@osherdp WDYT?

Reply:
``created[after]`` and ``created[before]`` parameters seem to work pretty well to me
why not use them and be explicit about the time-frame?

Reply:
We can do that, so if scraper scraps every hour, do we want to scrap last hour's usages every time?

Reply:
Sounds too risky I think, because we can skip some data this way
But scraping for the last 24 hours jobs sounds ok, and it shouldn't handle too much data

Reply:
Fixed